javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,264 @@
+const textarea = document.querySelector("textarea");
+const ul = document.querySelector("ul");
+let noteId = 0;
+
+document.querySelector("button.add-note").addEventListener("click", () => {
+    saveNote(textarea.value, noteId);
+    noteId++;
+    textarea.value = "";
+
+    // Clear the ul
+    ul.innerHTML = "";
+    notes.forEach((note) => {
+        const li = document.createElement("li");
+        li.innerHTML = note.content;
+
+        // ! ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+        // I think these would make the app better,
+        // if a note is important, the user can make its background red by clicking on it,
+        // also if a note is done, the user by double-clicking on it will make it line-through
+
+        // To make a note Important by adding a class with red background color
+        li.onclick = () => li.classList.toggle("isImportant");
+
+        // To make a note Done by adding a class with line-through style
+        li.ondblclick = () => li.classList.toggle("isDone");
+        // ! ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑
+
+        ul.appendChild(li);
+    });
+});
+
+document
+    .querySelector("button.log-out")
+    .addEventListener("click", logOutNotesFormatted);
+
+console.log(
+    "-----------------------------------------------------------------"
+);
+
+// CactusIO-interactive (Smart phone usage app) optional
+
+// Adding an activity
+
+// Initializing an empty array for restoring the activities
+let activities = [];
+
+// To get the activity day in this format: 2/30/2024
+const today = new Date();
+const activityDay = today.toLocaleDateString();
+
+// To create an object from any activity and then adding it to the activities array
+function addActivity(activity, duration) {
+    activities.push({
+        date: activityDay,
+        activity: activity,
+        duration: duration,
+    });
+}
+
+addActivity("Youtube", 30);
+addActivity("Instagram", 40);
+addActivity("Facebook", 8);
+addActivity("Gaming", 20);
+
+// Show my status
+
+// Default value for usage limit
+let usageLimit = 90;
+
+// To check the total time of the activities and compare it with the specified limit value
+function showStatus(activities) {
+    if (activities.length === 0) {
+        return "Add some activities before calling showStatus";
+    }
+    // The initial value of the variable that will accumulate the durations of the activities
+    let totalDuration = 0;
+
+    for (let activity of activities) {
+        totalDuration += activity.duration;
+    }
+
+    let status = `You have added ${activities.length} activities. They amount to ${totalDuration} min. of usage`;
+
+    if (totalDuration > usageLimit) {
+        status +=
+            "\nYou have reached your limit, no more smart-phoning for you!";
+    }
+
+    return status;
+}
+
+console.log(showStatus(activities));
+
+// To find the activity that the user spent the most time on
+function mostTimeActivity(activities) {
```

Comment:
Good idea for expanding the solution.

Reply:
Thank you dear Jason for reviewing my code, 👍 ❤️